src/__tests__/yieldSources.test.ts

```diff
@@ -0,0 +1,72 @@
+import { getKnownYieldSourceContract } from '../yieldSources'
+import { YIELD_SOURCE_NAMES } from '../data/knownYieldSources'
+
+describe('getKnownYieldSourceContract', () => {
+  it('returns a KnownYieldSourceContract if matches', () => {
+    const knownContract = getKnownYieldSourceContract(
+      1,
+      '0x39AA39c021dfbaE8faC545936693aC917d5E7563'
+    )
+    expect(knownContract).toEqual({
+      option: {
+        image: '/tokens/usdc-new-transparent.png',
+        label: '$USDC - Compound Yield',
+        value: '0x39AA39c021dfbaE8faC545936693aC917d5E7563'
+      },
+      type: 'compound',
+      yieldSourceName: YIELD_SOURCE_NAMES.compound
+    })
+  })
+
+  it('supports any case sensitivity', () => {
+    const knownContract = getKnownYieldSourceContract(
+      1,
+      '0xF8445C529D363CE114148662387EBA5E62016E20'
+    )
+    expect(knownContract).toEqual({
+      option: {
+        image: '/tokens/rai-small.png',
+        label: '$RAI - CREAM Yield',
+        value: '0xf8445C529D363cE114148662387eba5E62016e20'
+      },
+      type: 'compound',
+      yieldSourceName: YIELD_SOURCE_NAMES.cream
+    })
+  })
+
+  it('supports other networks', () => {
+    const knownContract = getKnownYieldSourceContract(
+      56,
+      '0xC17C8C5B8BB9456C624F8534FDE6CBDA2451488C'
+    )
+    expect(knownContract).toEqual({
+      option: {
+        image: '/tokens/iotx-small.svg',
+        label: '$IOTX - CREAM Yield',
+        value: '0xc17C8C5b8bB9456c624f8534FdE6cBda2451488C'
+      },
+      type: 'compound',
+      yieldSourceName: YIELD_SOURCE_NAMES.cream
+    })
+  })
+
+  it('returns undefined if no matches', () => {
+    const knownContract = getKnownYieldSourceContract(
+      1,
+      '0x6b175474e89094c44da98b954eedeac495271d0f'
+    )
+    expect(knownContract).toEqual(undefined)
+  })
+
+  it('throws with bad params', () => {
+    expect(() => {
+      getKnownYieldSourceContract()
```

Comment:
The tests are failing for me here:

```
src/__tests__/yieldSources.test.ts:63:7 - error TS2554: Expected 2 arguments, but got 0.

    63       getKnownYieldSourceContract()
             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

      src/yieldSources.ts:21:3
        21   chainId: number,
             ~~~~~~~~~~~~~~~
        An argument for 'chainId' was not provided.
```

Reply:
oh yah, the TS strictness issue. I thought I commented that test out? Maybe forgot to push.

I looked it up this morn and I think I can pass `(chainId: any, address: any)` for undefined values.